src/src/tx_main.cpp

```diff
@@ -389,6 +390,47 @@ void ICACHE_RAM_ATTR HandlePrepareForTLM()
   }
 }
 
+int8_t ICACHE_RAM_ATTR PowerEnumToLBTLimit(PowerLevels_e txPower)
+{
+    // Calculated from EN 300 328, assuming 800kHz BW for sx1280
+    // TL = -70 dBm/MHz + 10 × log10 (100 mW / Pout) (Pout in mW e.i.r.p.)
+    // Values above 100mW are not relevant, default to 100mW threshold
+    // TODO: This threshold should be modified with a config for antenna gain
+    switch(txPower)
+    {
+    case PWR_10mW: return -78;
+    case PWR_25mW: return -82;
+    case PWR_50mW: return -84;
+    case PWR_100mW: return -88;
+    default: return -88;
+    }
+}
+
+volatile uint32_t RxStartTimeCCA = 0;
+
+void ICACHE_RAM_ATTR BeginClearChannelAssessment()
+{
+  // Listen Before Talk (LBT) aka clear channel assessment (CCA)
+  // Start RX early because it takes about 90us from RX enable to 
+  // valid instant RSSI values are returned.
+  // Not interested in packets or interrupts while measuring RF energy on channel.
+
+  Radio.SetDioIrqParams(SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
+  Radio.RXnb();
+  RxStartTimeCCA = micros();
+}
+
+bool ICACHE_RAM_ATTR ChannelIsClear()
+{
+  while(micros() - RxStartTimeCCA < ExpressLRS_currAirRate_RFperfParams->InstantRssiValidDelayUs);
+  int8_t rssiResult = Radio.GetRssiInst();
+
+  Radio.SetTxIdleMode();
+  Radio.ClearIrqStatus(SX1280_IRQ_RADIO_ALL);
+  Radio.SetDioIrqParams(SX1280_IRQ_RADIO_ALL, SX1280_IRQ_TX_DONE | SX1280_IRQ_RX_DONE, SX1280_IRQ_RADIO_NONE, SX1280_IRQ_RADIO_NONE);
+  return rssiResult < PowerEnumToLBTLimit((PowerLevels_e)config.GetPower());
```

Comment:
If the LBT threshold should dynamically adapt to the current power level then the argument here should be `POWERMGNT.currPower()`. `(PowerLevels_e)config.GetPower()` returns the configured power level = the max power level when dynamic power is enabled.

Reply:
Good catch, this LBT feature should be tested for compliance in fixed 100mW power mode, but currPower seems correct here, thanks!